ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1650,7 +1649,7 @@ private InstallSnapshotReplyProto notifyStateMachineToInstallSnapshot(
       }
       changeToFollowerAndPersistMetadata(leaderTerm, "installSnapshot");
       state.setLeader(leaderId, "installSnapshot");
-      long snapshotIndex = state.getSnapshotIndex();
+      long snapshotIndex = state.getLog().getSnapshotIndex();
```

Comment:
maybe we can put this line before `if (snapshotIndex + 1 >= firstAvailableLogIndex && firstAvailableLogIndex > 0)`

Reply:
Since it is protected by the descriptor ```synchronized```, it should be fine here